src/main/java/org/folio/service/inventory/InventoryItemStatusSyncService.java

```diff
@@ -0,0 +1,147 @@
+package org.folio.service.inventory;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import one.util.streamex.StreamEx;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.completablefuture.VertxFutureRepeater;
+import org.folio.models.ItemStatus;
+import org.folio.okapi.common.GenericCompositeFuture;
+import org.folio.orders.utils.PoLineCommonUtil;
+import org.folio.orders.utils.RequestContextUtil;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.jaxrs.model.Location;
+import org.folio.rest.jaxrs.model.PoLine;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.folio.helper.BaseHelper.MAX_REPEAT_ON_FAILURE;
+import static org.folio.rest.RestConstants.MAX_IDS_FOR_GET_RQ_15;
+
+public class InventoryItemStatusSyncService {
```

Comment:
Do you agree that methods like `updateItemWithPieceFields` and baisaclly any `InventoryItemManager` method which is to complex and more than CRUD is a candidate for moving in this class?

Reply:
I just did not want to couple new service to Piece entity, in this case we can use InventoryItemStatusSyncService in places where pieces are not presented.